api_fhir_r4/views.py

```diff
@@ -40,66 +45,130 @@ class BaseFHIRView(APIView):
+        refDate = request.GET.get('refDate')
         claim_date = request.GET.get('claimDateFrom')
         identifier = request.GET.get("identifier")
-
-        queryset = self.get_queryset()
-
-        if claim_date is not None:
-            day, month, year = claim_date.split('-')
-            try:
-                claim_date_parsed = datetime.datetime(int(year), int(month), int(day))
-            except ValueError:
-                result = OperationOutcomeConverter.build_for_400_bad_request("claimDateFrom should be in dd-mm-yyyy format")
-                return Response(result.toDict(), status.HTTP_400_BAD_REQUEST)
-            has_claim_in_range = Claim.objects\
-                .filter(date_claimed__gte=claim_date_parsed)\
-                .filter(insuree_id=OuterRef("id"))\
-                .values("id")
-            queryset = queryset.annotate(has_claim_in_range=Exists(has_claim_in_range)).filter(has_claim_in_range=True)
-
         if identifier:
             queryset = queryset.filter(chf_id=identifier)
+        else:
+            queryset = queryset.filter(validity_to__isnull=True).order_by('validity_from')
+            if refDate != None:
+                # isValidDate = True
+                try :
+                    datevar = datetime.datetime.strptime(refDate, "%Y-%m-%d").date()
+                except ValueError :
+                    pass
+                    # isValidDate = False
+                #datevar = refDate
+                queryset = queryset.filter(validity_from__gte=datevar)
+            if claim_date is not None:
+                #year,month,day = refDate.split('-')
+                try:
+                    claim_parse_dated = datetime.datetime.strptime(claim_date, "%Y-%m-%d").date()
+                    #datetime.datetime(int(year), int(month), int(day))
+                except ValueError:
+                    result = OperationOutcomeConverter.build_for_400_bad_request("claimDateFrom should be in dd-mm-yyyy format")
+                    return Response(result.toDict(), status.HTTP_400_BAD_REQUEST)
+                has_claim_in_range = Claim.objects\
+                    .filter(date_claimed__gte=claim_parse_dated)\
+                    .filter(insuree_id=OuterRef("id"))\
+                    .values("id")
+                queryset = queryset.annotate(has_claim_in_range=Exists(has_claim_in_range)).filter(has_claim_in_range=True)
 
         serializer = PatientSerializer(self.paginate_queryset(queryset), many=True)
         return self.get_paginated_response(serializer.data)
 
+    
     def get_queryset(self):
-        return Insuree.get_queryset(None, self.request.user)
+        return Insuree.objects
+    
+class GroupViewSet(BaseFHIRView, viewsets.ModelViewSet):
+    lookup_field = 'uuid'
+    serializer_class = GroupSerializer
+
+    def list(self, request, *args, **kwargs):
+        queryset = self.get_queryset()
+        serializer = GroupSerializer(self.paginate_queryset(queryset), many=True)
```

Comment:
@davismutinda  need to had validityto__isnull= True, when a list is requested, Also please support the Identifier case